tests/integration/test_k8s.py FAILED

From https://ci.centos.org/job/user-cont-conu/19/consoleFull

```
14:53:28 tests/integration/test_k8s.py::test_pod FAILED                           [ 50%]
14:53:53 tests/integration/test_k8s.py::test_database_deployment FAILED           [ 51%]
```

```python
14:54:34 =================================== FAILURES ===================================
14:54:34 ___________________________________ test_pod ___________________________________
14:54:34 
14:54:34     def test_pod():
14:54:34         with DockerBackend() as backend:
14:54:34             image = backend.ImageClass(FEDORA_MINIMAL_REPOSITORY, tag=FEDORA_MINIMAL_REPOSITORY_TAG)
14:54:34     
14:54:34 >           pod = image.run_in_pod(namespace='default')
14:54:34 
14:54:34 tests/integration/test_k8s.py:33: 
14:54:34 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
14:54:34 
14:54:34 self = DockerImage(repository=registry.fedoraproject.org/fedora-minimal, tag=26)
14:54:34 namespace = 'default'
14:54:34 
14:54:34     def run_in_pod(self, namespace="default"):
14:54:34         """
14:54:34             run image inside Kubernetes Pod
14:54:34             :param namespace: str, name of namespace where pod will be created
14:54:34             :return: Pod instance
14:54:34             """
14:54:34     
14:54:34         image_data = self.get_metadata()
14:54:34     
14:54:34         pod = Pod.create(image_data)
14:54:34     
14:54:34         try:
14:54:34             pod_instance = api.create_namespaced_pod(namespace=namespace, body=pod)
14:54:34         except ApiException as e:
14:54:34 >           raise ConuException("Exception when calling CoreV1Api->create_namespaced_pod: %s\n" % e)
14:54:34 E           ConuException: Exception when calling CoreV1Api->create_namespaced_pod: (403)
14:54:34 E           Reason: Forbidden
14:54:34 E           HTTP response headers: HTTPHeaderDict({'Date': 'Fri, 22 Jun 2018 14:53:28 GMT', 'Content-Length': '287', 'Content-Type': 'application/json', 'X-Content-Type-Options': 'nosniff', 'Cache-Control': 'no-store'})
14:54:34 E           HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"pods is forbidden: User \"developer\" cannot create pods in the namespace \"default\": User \"developer\" cannot create pods in project \"default\"","reason":"Forbidden","details":{"kind":"pods"},"code":403}
```

```python
14:54:34 ___________________________ test_database_deployment ___________________________
14:54:34 
14:54:34     def test_database_deployment():
14:54:34         with DockerBackend() as backend:
14:54:34             postgres_image = backend.ImageClass("centos/postgresql-10-centos7")
14:54:34     
14:54:34             postgres_image_metadata = postgres_image.get_metadata()
14:54:34     
14:54:34             # set up env variables
14:54:34     
14:54:34             db_env_variables = {"POSTGRESQL_USER": "user",
14:54:34                                 "POSTGRESQL_PASSWORD": "pass",
14:54:34                                 "POSTGRESQL_DATABASE": "db"}
14:54:34     
14:54:34             postgres_image_metadata.env_variables.update(db_env_variables)
14:54:34     
14:54:34             db_labels = {"app": "postgres"}
14:54:34     
14:54:34 >           db_service = Service(name="database", ports=["5432"], selector=db_labels)
14:54:34 
14:54:34 tests/integration/test_k8s.py:59: 
14:54:34 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
14:54:34 
14:54:34 self = <conu.backend.k8s.service.Service object at 0x7fd8b0276ad0>
14:54:34 name = 'database', ports = ['5432'], namespace = 'default', labels = None
14:54:34 selector = {'app': 'postgres'}
14:54:34 
14:54:34     def __init__(self, name, ports, namespace='default', labels=None, selector=None):
14:54:34         """
14:54:34             Utility functions for kubernetes services.
14:54:34     
14:54:34             :param name: str, name of the service
14:54:34             :param namespace: str, name of the namespace
14:54:34             :param ports: list of str, list of exposed ports, example:
14:54:34                     - ['1234/tcp', '8080/udp']
14:54:34             :param labels: dict, dict of labels
14:54:34             :param selector: dict, route service traffic to pods with label keys and
14:54:34              values matching this selector
14:54:34             """
14:54:34         self.name = name
14:54:34         self.namespace = namespace
14:54:34         self.ports = ports
14:54:34     
14:54:34         exposed_ports = []
14:54:34     
14:54:34         # create Kubernetes service port objects
14:54:34         for port in self.ports:
14:54:34             splits = port.split("/", 1)
14:54:34             port = int(splits[0])
14:54:34             protocol = splits[1].upper() if len(splits) > 1 else None
14:54:34             exposed_ports.append(client.V1ServicePort(port=port, protocol=protocol))
14:54:34     
14:54:34         metadata = client.V1ObjectMeta(name=self.name, namespace=self.namespace, labels=labels)
14:54:34         self.spec = client.V1ServiceSpec(ports=exposed_ports, selector=selector)
14:54:34     
14:54:34         body = client.V1Service(spec=self.spec, metadata=metadata)
14:54:34     
14:54:34         try:
14:54:34             api.create_namespaced_service(self.namespace, body)
14:54:34         except ApiException as e:
14:54:34             raise ConuException(
14:54:34 >               "Exception when calling Kubernetes API - create_namespaced_service: {}\n".format(e))
14:54:34 E           ConuException: Exception when calling Kubernetes API - create_namespaced_service: (403)
14:54:34 E           Reason: Forbidden
14:54:34 E           HTTP response headers: HTTPHeaderDict({'Date': 'Fri, 22 Jun 2018 14:53:53 GMT', 'Content-Length': '303', 'Content-Type': 'application/json', 'X-Content-Type-Options': 'nosniff', 'Cache-Control': 'no-store'})
14:54:34 E           HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"services is forbidden: User \"developer\" cannot create services in the namespace \"default\": User \"developer\" cannot create services in project \"default\"","reason":"Forbidden","details":{"kind":"services"},"code":403}
```

--- a/Makefile
+++ b/Makefile
@@ -45,13 +45,10 @@ test: build-test-container test-in-container test-doc-examples
 centos-ci-test: install-test-requirements build-test-container test-in-container
 
 test-in-container:
-	@# use it like this: `make test-in-container TEST_TARGET=tests/integration/test_utils.py`k
-	docker run --net=host --rm -v /dev:/dev:ro -v /var/lib/docker:/var/lib/docker:ro --security-opt label=disable --cap-add SYS_ADMIN -e KUBECONFIG=/var/lib/origin/openshift.local.config/master/admin.kubeconfig -v /var/lib/origin/openshift.local.config/master/admin.kubeconfig:/var/lib/origin/openshift.local.config/master/admin.kubeconfig -ti -v /var/run/docker.sock:/var/run/docker.sock -v $(CURDIR):/src -v $(CURDIR)/pytest-container.ini:/src/pytest.ini $(TEST_IMAGE_NAME) /bin/bash -c "oc login 127.0.0.1:8443 -u system:admin --config=/var/lib/origin/openshift.local.config/master/admin.kubeconfig --insecure-skip-tls-verify && make exec-test TEST_TARGET=$(TEST_TARGET)"
-
-test-k8s-minikube: build-test-container
+	@# use it like this: `make test-in-container TEST_TARGET=tests/integration/test_utils.py`
 	$(eval kubedir := $(shell mktemp -d /tmp/tmp.conu-kube-XXXXX))
 	sed -e s#"${HOME}"#/root#g ${HOME}/.kube/config > $(kubedir)/config
-	docker run --net=host --rm -v /dev:/dev:ro -v /var/lib/docker:/var/lib/docker:ro --security-opt label=disable --cap-add SYS_ADMIN -ti -v /var/run/docker.sock:/var/run/docker.sock -v $(CURDIR):/src -v $(CURDIR)/pytest-container.ini:/src/pytest.ini -v ${HOME}/.minikube:/root/.minikube -v $(kubedir):/root/.kube $(TEST_IMAGE_NAME) make exec-test TEST_TARGET=tests/integration/test_k8s.py
+	docker run --net=host --rm -v /dev:/dev:ro -v /var/lib/docker:/var/lib/docker:ro --security-opt label=disable --cap-add SYS_ADMIN -ti -v /var/run/docker.sock:/var/run/docker.sock -v $(CURDIR):/src -v $(CURDIR)/pytest-container.ini:/src/pytest.ini -v ${HOME}/.minikube:/root/.minikube -v $(kubedir):/root/.kube $(TEST_IMAGE_NAME) make exec-test
 
 test-in-vm:
 	vagrant up --provision

--- a/test-requirements.sh
+++ b/test-requirements.sh
@@ -7,12 +7,23 @@ source /etc/os-release
 set -x
 
 if [ "${NAME}" == "Fedora" ]; then
-    dnf install -y docker origin-clients make
+    dnf install -y docker make
 elif [ "${NAME}" == "CentOS Linux" ]; then
-    yum install -y centos-release-openshift-origin39
-    yum install -y docker origin-clients make
+    yum install -y docker make
 fi
 
+setenforce 0
 systemctl start docker
-oc cluster status || oc cluster up --skip-registry-check=true
-oc get project conu || oc new-project conu
+curl -Lo minikube https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64 && chmod +x minikube
+curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/v1.10.0/bin/linux/amd64/kubectl && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
+
+
+export MINIKUBE_WANTUPDATENOTIFICATION=false
+export MINIKUBE_WANTREPORTERRORPROMPT=false
+export MINIKUBE_HOME=$HOME
+export CHANGE_MINIKUBE_NONE_USER=true
+mkdir -p $HOME/.kube
+touch $HOME/.kube/config
+
+export KUBECONFIG=$HOME/.kube/config
+./minikube start --vm-driver=none --extra-config=apiserver.admission-control="" --extra-config=kubelet.cgroup-driver=systemd

--- a/test-requirements.sh
+++ b/test-requirements.sh
@@ -15,7 +15,7 @@ fi
 setenforce 0
 systemctl start docker
 curl -Lo minikube https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64 && chmod +x minikube
-curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/v1.10.0/bin/linux/amd64/kubectl && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
+curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/v1.10.0/bin/linux/amd64/kubectl && chmod +x kubectl && mv kubectl /usr/local/bin/
 
 
 export MINIKUBE_WANTUPDATENOTIFICATION=false
@@ -26,4 +26,4 @@ mkdir -p $HOME/.kube
 touch $HOME/.kube/config
 
 export KUBECONFIG=$HOME/.kube/config
-./minikube start --vm-driver=none --extra-config=apiserver.admission-control="" --extra-config=kubelet.cgroup-driver=systemd
+./minikube start --vm-driver=none --extra-config=apiserver.admission-control="" --extra-config=kubelet.cgroup-driver=systemd --kubernetes-version=v1.10.0

--- a/tests/integration/test_k8s.py
+++ b/tests/integration/test_k8s.py
@@ -18,8 +18,6 @@
 Tests for Kubernetes backend
 """
 
-import pytest
-
 from conu import DockerBackend
 from conu.backend.k8s.backend import K8sBackend
 from conu.backend.k8s.pod import PodPhase
@@ -32,7 +30,6 @@ import urllib3
 urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
 
 
-@pytest.mark.xfail
 def test_pod():
     with K8sBackend() as k8s_backend:
 
@@ -52,7 +49,6 @@ def test_pod():
                 k8s_backend.delete_namespace(namespace)
 
 
-@pytest.mark.xfail
 def test_database_deployment():
     with K8sBackend() as k8s_backend:
 
@@ -90,7 +86,6 @@ def test_database_deployment():
                 k8s_backend.delete_namespace(namespace)
 
 
-@pytest.mark.xfail
 def test_list_pods():
     with K8sBackend() as k8s_backend:
 
@@ -127,7 +122,6 @@ def test_list_services():
             k8s_backend.delete_namespace(namespace)
 
 
-@pytest.mark.xfail
 def test_list_deployments():
     with K8sBackend() as k8s_backend:
 

